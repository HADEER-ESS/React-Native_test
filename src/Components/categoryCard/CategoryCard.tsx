import React from 'react'
import taskCategory from '../../Types/TaskCategory'
import { StyleSheet, Text, TouchableOpacity, View } from 'react-native'

const CategoryCard = (props: { selectedItem: string, category: taskCategory, getTaskType: (value: string) => void }) => {
  return (
    <TouchableOpacity onPress={() => {
      props.getTaskType(props?.category?.type)
    }}
      testID='task_category_click'
      style={[style.cardCategoryContainer, { backgroundColor: props.category.type == props.selectedItem ? "#DFF5FF" : "transparent" }]}
    >

      <View style={[style.viewTextStyle, { backgroundColor: props?.category?.color }]}></View>

      <Text style={style.textStyle}>{props?.category?.type}</Text>

    </TouchableOpacity>
  )
}

export default CategoryCard

const style = StyleSheet.create({
  cardCategoryContainer: {
    flexDirection: 'row',
    marginTop: 16,
    height: "15%",
    alignItems: 'center',
    paddingHorizontal: 4,
    marginHorizontal: 4,
    borderRadius: 14
  },
  viewTextStyle: {
    width: 16,
    height: 16,
    borderRadius: 20
  },
  textStyle: {
    color: '#000000',
    paddingHorizontal: 3
  }
})