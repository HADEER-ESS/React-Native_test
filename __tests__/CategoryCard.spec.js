import "react-native";
import React from "react";
import CategoryCard from "../src/Components/categoryCard/CategoryCard";
import { fireEvent, render } from "@testing-library/react-native";


describe("<CategoryCard />", () => {
    test("Display the component", () => {

        let item = {
            type: "Urgent",
            color: 'red'
        }

        let taskType = "Important"

        let addTaskFun = jest.fn()

        const tree = render(<CategoryCard category={item} getTaskType={addTaskFun} selectedItem={taskType} />)


        //toMatchInlineSnapshot cause this error ==> Jest: Inline Snapshots are not supported when using Prettier 3.0.0 or above
        //need to downgrade Prettier till fix this issue in upgrade

        expect(tree).toMatchSnapshot()
    })

    test("item Clicked and select one task category", () => {
        let addTaskFun = jest.fn()
        let item = {
            type: "Urgent",
            color: 'red'
        }

        const { getByTestId } = render(<CategoryCard category={item} getTaskType={addTaskFun} />)

        const pressable = getByTestId('task_category_click')

        fireEvent.press(pressable)
        expect(addTaskFun).toHaveBeenCalled()

    })
})

