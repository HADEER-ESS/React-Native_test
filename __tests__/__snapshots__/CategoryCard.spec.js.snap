// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryCard /> Display the component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 14,
      "flexDirection": "row",
      "height": "15%",
      "marginHorizontal": 4,
      "marginTop": 16,
      "opacity": 1,
      "paddingHorizontal": 4,
    }
  }
  testID="task_category_click"
>
  <View
    style={
      [
        {
          "borderRadius": 20,
          "height": 16,
          "width": 16,
        },
        {
          "backgroundColor": "red",
        },
      ]
    }
  />
  <Text
    style={
      {
        "color": "#000000",
        "paddingHorizontal": 3,
      }
    }
  >
    Urgent
  </Text>
</View>
`;

exports[`Display the component 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "transparent",
      "borderRadius": 14,
      "flexDirection": "row",
      "height": "15%",
      "marginHorizontal": 4,
      "marginTop": 16,
      "opacity": 1,
      "paddingHorizontal": 4,
    }
  }
>
  <View
    style={
      {
        "backgroundColor": "red",
        "borderRadius": 20,
        "height": 16,
        "width": 16,
      }
    }
  />
  <Text
    style={
      {
        "color": "#000000",
        "paddingHorizontal": 3,
      }
    }
  >
    Urgent
  </Text>
</View>
`;
